modules/data/redis-cache/main.bicep

```diff
@@ -0,0 +1,250 @@
+param redisConfiguration object = {}
+
+@allowed([
+  '4.0'
+  '6.0'
+])
+@description('Optional. Redis version. Only major version will be used in PUT/PATCH request with current valid values: (4, 6).')
+param redisVersion string = '6.0'
+
+@allowed([
+  0
+  1
+  2
+  3
+  4
+  5
+  6
+])
+@description('Optional. The size of the Redis cache to deploy. Valid values: for C (Basic/Standard) family (0, 1, 2, 3, 4, 5, 6), for P (Premium) family (1, 2, 3, 4).')
+param capacity int = 1
+
+@minValue(1)
+@description('Optional. The number of shards to be created on a Premium Cluster Cache.')
+param shardCount int = 1
+
+@minValue(1)
+@description('Optional. Amount of replicas to create per master for this Redis Cache.')
+param replicasPerMaster int = 1
+
+@minValue(1)
+@description('Optional. Amount of replicas to create per primary for this Redis Cache.')
+param replicasPerPrimary int = 1
+
+
+@description('Optional. Static IP address. Optionally, may be specified when deploying a Redis cache inside an existing Azure Virtual Network; auto assigned by default.')
+param staticIP string = ''
+
+@description('Optional. The full resource ID of a subnet in a virtual network to deploy the Redis cache in. Example format: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/Microsoft.{Network|ClassicNetwork}/VirtualNetworks/vnet1/subnets/subnet1.')
+param subnetId string = ''
+
+@description('Optional. A dictionary of tenant settings.')
+param tenantSettings object = {}
+
+@description('Specifies the number of days that logs will be kept for; a value of 0 will retain data indefinitely.')
+@minValue(0)
+@maxValue(365)
+param diagnosticLogsRetentionInDays int = 365
+
+@description('Resource ID of the diagnostic storage account.')
+param diagnosticStorageAccountId string = ''
+
+@description('Resource ID of the diagnostic log analytics workspace.')
+param diagnosticWorkspaceId string = ''
+
+@description('Resource ID of the diagnostic event hub authorization rule for the Event Hubs namespace in which the event hub should be created or streamed to.')
+param diagnosticEventHubAuthorizationRuleId string = ''
+
+@description('Name of the diagnostic event hub within the namespace to which logs are streamed. Without this, an event hub is created for each log category.')
+param diagnosticEventHubName string = ''
+
+@description('The name of logs that will be streamed.')
+@allowed([
+  'ConnectedClientList'
+])
+param logsToEnable array = [
+  'ConnectedClientList'
+]
+
+@description('The name of metrics that will be streamed.')
+@allowed([
+  'AllMetrics'
+])
```

Comment:
```suggestion
@description('The name of metrics that will be streamed. https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/metrics-supported#microsoftcacheredis')
```

Reply:
let's let the user provide additional values if they would like to here, and provide a pointer to the list.

Reply:
Currently supported categories are: 'AllMetrics' , There is no any other value is allowed, User can not provide any other additional values.